### PR TITLE
add intoto support to rekor client

### PR DESCRIPTION
**Pull request by bdehamer**
#### Summary
Updates the `Rekor` client to support "intoto" entry uploads. Previously, the only entry kind that the client knew how to deal with was "hashrekord". The changes here lay the groundwork for supporting all of the various entry kinds in the future but is primarily focused on the "hashrekord" and "intoto" kinds for now.

#### Release Note

* Updates the `Rekor.createEntry` function to support creating entries of any kind.
* Adds a new `Rekor.createIntotoEntry` convenience function.
* Adds a new `Rekor.createHashedRekordEntry` convenience function.
* Renames the `Rekor.searchLog` function to `Rekor.searchIndex` to better reflect which API is being wrapped.

### CODE REVIEW

File: src/client/rekor.test.ts [hunk 1]
```diff
@@ -89,6 +102,90 @@ describe('Rekor', () => {
...
+    const artifactDigest =
+      '1c025a6e48ceb8bf10e01b367089732326eabe3541d03d348724c79040382c65';
+    const artifactSignature =
+      'MEUCIDB2SWDabztSC8RrlfRCWUf04LBN0E2CEwiDZJLacDS8AiEA3bQHMBpodxA3dvJ+JK1SALkuzju/w4oCg3S89c8CtN8=';
+    const publicKey =
+      'LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K';
+
+    const proposedEntry = {
+      apiVersion: '0.0.1',
+      kind: 'hashedrekord',
```
Comment (feelepxyz) [inline src/client/rekor.test.ts:115]: Should this be intoto? Looks the test case below expects it to submit a new intoto entry

```suggestion
      kind: 'intoto',
```

**Review by feelepxyz — DISMISSED** [review]
LGTM 👍  Added a minor test comment

File: src/client/rekor.test.ts (hunk 1 above)
Comment (bdehamer) [reply src/client/rekor.test.ts:115]: Good catch -- it was actually that I'd reversed the test descriptions in the `describe` string. Fixed now.

Comment (feelepxyz) [reply src/client/rekor.test.ts:115]: Did you push your changes? Looks like the below `it` still reads "submits a new intoto entry"

**Review by feelepxyz — APPROVED** [review]

File: src/client/rekor.test.ts [hunk 2]
```diff
@@ -89,6 +102,90 @@ describe('Rekor', () => {
...
+
+    beforeEach(() => {
+      nock(baseURL)
+        .matchHeader('Accept', 'application/json')
+        .matchHeader('Content-Type', 'application/json')
+        .post('/api/v1/log/entries', proposedEntry)
+        .reply(201, responseBody);
+    });
+
+    it('submits a new intoto entry', async () => {
```
Comment (feelepxyz) [inline src/client/rekor.test.ts:146]:
```suggestion
    it('submits a new hashedrekor entry', async () => {
```

File: src/client/rekor.test.ts [hunk 3]
```diff
@@ -89,6 +102,90 @@ describe('Rekor', () => {
...
+
+    beforeEach(() => {
+      nock(baseURL)
+        .matchHeader('Accept', 'application/json')
+        .matchHeader('Content-Type', 'application/json')
+        .post('/api/v1/log/entries', proposedEntry)
+        .reply(201, responseBody);
+    });
+
+    it('submits a new hashedrekor entry', async () => {
```
Comment (feelepxyz) [inline src/client/rekor.test.ts:183]:
```suggestion
    it('submits a new intoto entry', async () => {
```

**Review by feelepxyz — APPROVED** [review]

**Pull request merged**